app/cdap/components/ConnectionList/index.tsx

```diff
@@ -198,76 +280,138 @@ export default function ConnectionList() {
+                } else {
+                  headerContent = (
+                    <>
+                      <Box
+                        className={
+                          eachFilteredData.isSearching
+                            ? classes.hideComponent
+                            : classes.beforeSearchIconClickDisplay
+                        }
+                      >
+                        {headersRefs.current[index]?.offsetWidth <
+                        headersRefs.current[index]?.scrollWidth ? (
+                          <CustomTooltip title={dataForTabs[index - 1].selectedTab} arrow>
+                            <Typography
+                              variant="body2"
+                              ref={(element) => {
+                                headersRefs.current[index] = element;
+                              }}
+                            >
+                              {filteredData[index - 1].selectedTab}
+                            </Typography>
+                          </CustomTooltip>
+                        ) : (
+                          <Typography
+                            variant="body2"
+                            ref={(element) => {
+                              headersRefs.current[index] = element;
+                            }}
+                          >
+                            {filteredData[index - 1].selectedTab}
+                          </Typography>
+                        )}
+
+                        <Box
+                          onClick={() => {
+                            searchHandler(index);
+                          }}
+                        >
+                          <IconButton>
+                            <SearchRoundedIcon />
+                          </IconButton>
+                        </Box>
+                      </Box>
+                      <Box
+                        className={
+                          eachFilteredData.isSearching
+                            ? classes.afterSearchIconClick
+                            : classes.hideComponent
+                        }
+                        onMouseOver={() => makeCursorFocused(index)}
                       >
-                        {dataForTabs[index - 1].selectedTab}
-                      </Typography>
-                    </Box>
-                  </CustomTooltip>
-                ) : (
-                  <Box className={classes.beforeSearchIconClickDisplay}>
-                    <Typography
-                      variant="body2"
-                      className={classes.headerLabel}
-                      ref={(element) => {
-                        refs.current[index] = element;
-                      }}
-                    >
-                      {dataForTabs[index - 1].selectedTab}
-                    </Typography>
+                        <SearchRoundedIcon />
+                        <input
+                          type="text"
```

Comment:
this can be a common component
that can we used across the app

Reply:
Done